### PR TITLE
Do not filtering out conversation with @ in name

### PR DESCRIPTION
**Pull request by marcoconti83**
# Reason for this pull request
When filtering out conversation from a search result when the query was a username search (with starting with `@`), conversations that have `@` in the name should still be included

### CODE REVIEW

File: Source/UserSession/Search/Internal/ZMSearch.m [hunk 1]
```diff
@@ -155,15 +155,17 @@ - (NSArray *)connectedUsersMatchingSearchString:(NSString *)searchString
 
 - (NSArray *)conversationsMatchingSearchString:(NSString *)searchString
 {
-    if([searchString hasPrefix:@"@"]) { // ignore group conversations if query is for usernames
-        return @[];
-    }
     NSFetchRequest *conversationFetchRequest = [ZMConversation sortedFetchRequestWithPredicate:[ZMConversation predicateForSearchString:searchString]];
```
Comment (mikeger) [inline Source/UserSession/Search/Internal/ZMSearch.m]: I think we can use different predicate in case search contains @: do not search in participant's names then. Would that eliminate the need in re-filtering the results?

Comment (daehn) [inline Source/UserSession/Search/Internal/ZMSearch.m]: We still want to include names that start with an @ / match the query

Comment (mikeger) [inline Source/UserSession/Search/Internal/ZMSearch.m]: @daehn right now we have a predicate `predicateForSearchString` to match either conversation or participant name. If query string contains '@' then if we match with this predicate, one or another (conv name or participant name) should have @. Then we don't need to extra filter the query result, as I understand?

Comment (marcoconti83) [inline Source/UserSession/Search/Internal/ZMSearch.m]: True but this predicate is set in data-model, and I did not want to put the logic of "if the user searched with a "@", then ..." there.

Comment (mikeger) [inline Source/UserSession/Search/Internal/ZMSearch.m]: Right, may be `predicateForSearchString` can have a parameter like `matchUserNames:Bool`, because when it matches usernames it does a subquery, would be cool to avoid doing the subquery and then filtering on top to remove the subquery effect.

Comment (marcoconti83) [inline Source/UserSession/Search/Internal/ZMSearch.m]: I would merge this in, then we can do that change in another PR as it requires changes in data model and the in-between frameworks

Comment (marcoconti83) [inline Source/UserSession/Search/Internal/ZMSearch.m]: Turns out we already have such a predicate, so no changes needed in data model. Will do another PR on SE.
